tests/testsuite.src/run_misc.at

```diff
@@ -14570,3 +14570,32 @@ TST-DECIMAL IS < ZERO-DECIMAL
 
 
 AT_CLEANUP
+
+AT_SETUP([Decimal constants working after sub-program call])
+AT_KEYWORDS([runmisc])
+
+# this used to cause a SIGSEGV
+
+AT_DATA([prog.cob], [
+       IDENTIFICATION DIVISION.
+       PROGRAM-ID. prog.
+       DATA DIVISION.
+       WORKING-STORAGE SECTION.
+       01  X  PIC 9(2).
+       PROCEDURE DIVISION.
+           CALL "prog2"
+           IF X + 42 <> 0
+               DISPLAY "OK".
```

Comment:
as a suggestion: consider using "`DISPLAY "OK" WITH NO ADVANCING"`, as that allows to use a single line AT_CHECK below